config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/ServiceEntityBuilder.java

```diff
@@ -36,60 +42,50 @@
+            buildAndAppendPropertiesElement(service.getProperties()
+                            .entrySet()
+                            .stream()
+                            .collect(Collectors
+                                    .toMap(p -> "property." + p.getKey(), p -> p.getKey().startsWith(PREFIX_ENV) ? "SERVICE_PROPERTY_" + p.getKey() : p.getValue())),
+                    document, serviceDetailElement);
         }
 
-        Element serviceElement = document.createElement("l7:Service");
-        serviceElement.setAttribute("id", id);
+        Element serviceElement = createElementWithAttribute(document, SERVICE, ATTRIBUTE_ID, id);
         serviceElement.appendChild(serviceDetailElement);
 
-        Element resourcesElement = document.createElement("l7:Resources");
-
-        Element resourceSetElement = document.createElement("l7:ResourceSet");
-        resourceSetElement.setAttribute("tag", "policy");
-
-        Element resourceElement = document.createElement("l7:Resource");
-        resourceElement.setAttribute("type", "policy");
-
+        Element resourcesElement = document.createElement(RESOURCES);
+        Element resourceSetElement = createElementWithAttribute(document, RESOURCE_SET, "tag", "policy");
+        Element resourceElement = createElementWithAttribute(document, RESOURCE, "type", "policy");
         resourceElement.setTextContent(documentFileUtils.elementToString(policy.getPolicyDocument()));
 
         resourceSetElement.appendChild(resourceElement);
         resourcesElement.appendChild(resourceSetElement);
         serviceElement.appendChild(resourcesElement);
-        return new Entity("SERVICE", policy.getName(), id, serviceElement);
+        return new Entity(SERVICE_TYPE, policy.getName(), id, serviceElement);
     }
 
     private Element buildServiceMappings(Service service) {
-        Element serviceMappingsElement = document.createElement("l7:ServiceMappings");
-        Element httpMappingElement = document.createElement("l7:HttpMapping");
+        Element serviceMappingsElement = document.createElement(SERVICE_MAPPINGS);
+        Element httpMappingElement = document.createElement(HTTP_MAPPING);
         serviceMappingsElement.appendChild(httpMappingElement);
-        Element urlPatternElement = document.createElement("l7:UrlPattern");
-        urlPatternElement.setTextContent(service.getUrl());
-        httpMappingElement.appendChild(urlPatternElement);
-        Element verbsElement = document.createElement("l7:Verbs");
+
+        httpMappingElement.appendChild(createElementWithTextContent(document, URL_PATTERN, service.getUrl()));
+        Element verbsElement = document.createElement(VERBS);
         service.getHttpMethods().forEach(method -> {
-            Element verb = document.createElement("l7:Verb");
+            Element verb = document.createElement(VERB);
```

Comment:
Should we do something like below, or is that overkill? lol      
```
  Element verbsElement = createElementWithChildren(
                document,
                VERBS,
                service.getHttpMethods()
                        .stream()
                        .map(method -> createElementWithTextContent(document, VERB, method))
                        .collect(Collectors.toList())
                        .toArray(new Element[service.getHttpMethods().size()]));
```

Reply:
I think it's going to be too complex in this case, I left some cases without change to avoid too many complicated code for so simple thing